src/pages/RegisterCardInfo/RegisterCardInfo.tsx

```diff
@@ -0,0 +1,72 @@
+import { useReducer, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import CardInfo from '../../components/CardInfo/CardInfo';
+import CardPreview from '../../components/CardPreview/CardPreview';
+import PaymentsBottomFixedButton from '../../components/common/PaymentsBottomFixedButton/PaymentsBottomFixedButton';
+
+import { MainContainer } from './RegisterCardInfo.styled';
+
+import cardInfoReducer from '../../store/cardInfoReducer';
+
+import OPTION from '../../constants/option';
+import PATH from '../../constants/path';
+
+const INITIAL_CARD_INFO_STATE: CardInfo = {
+  cardNumbers: { value: ['', '', '', ''], isComplete: false },
+  cardBrand: { value: 'none', isComplete: false },
+  cardCompany: { value: '', isComplete: false },
+  expiration: { value: ['', ''], isComplete: false },
+  name: { value: '', isComplete: false },
+  cvc: { value: '', isComplete: false },
+  password: { value: '', isComplete: false },
+}
+
+const COMPLETE_CONDITION = {
+  cardNumbers: (values: string[]) => values.every(value => value.length === OPTION.cardNumberMaxLength),
+  cardBrand: true,
+  cardCompany: (value: CardCompany) => (value as string).length !== 0,
+  expiration: (values: string[]) => values.every(value => value.length === OPTION.expirationDateMaxLength),
+  name: (value: string) => value.length !== 0,
+  cvc: (value: string) => value.length === OPTION.cvcMaxLength,
+  password: (value: string) => value.length === OPTION.passwordMaxLength,
+}
+
+const RegisterCardInfo = () => {
+  const [cardInfo, dispatchCardInfo] = useReducer(cardInfoReducer, INITIAL_CARD_INFO_STATE);
+  const [cardState, setCardState] = useState<CardState>('front');
+
+  const handleCardState = (cardState: CardState) => {
+    setCardState(cardState)
+  }
+
+  const showSubmitButton = (cardInfo: CardInfo): boolean => {
+    return (
+      COMPLETE_CONDITION.cardNumbers(cardInfo.cardNumbers.value) &&
+      COMPLETE_CONDITION.cardBrand &&
+      COMPLETE_CONDITION.cardCompany(cardInfo.cardCompany.value) &&
+      COMPLETE_CONDITION.expiration(cardInfo.expiration.value) &&
+      COMPLETE_CONDITION.name(cardInfo.name.value) &&
+      COMPLETE_CONDITION.cvc(cardInfo.cvc.value) &&
+      COMPLETE_CONDITION.password(cardInfo.password.value)
+    );
+  }
```

Comment:
COMPLETE_CONDITION이 일관되게 Record<string, (info: CardInfo) => boolean> 이렇게 관리되었으면 아래처럼 단순하게 작성할 수 있었을 것 같아요.
```suggestion
  const showSubmitButton = (cardInfo: CardInfo): boolean => {
    return Object.values(COMPLETE_CONDITION).every((condition) => condition(cardInfo))
  }
```

Reply:
오... 이렇게 사용할 수 있군요 코드가 굉장히 깔끔해 지는 느낌이네요!
아직 내장 메서드들의 활용을 많이 안해봐서 그런지, 이런 식의 생각은 어떻게 할 수 있는지 신기합니다... 하하

이해하고 적용시켜 보았습니다 :)

4198da7bd6ea32277a27e76aab27d59fc3ee7693